frontend/src/pages/checklist/[filters].tsx

```diff
@@ -99,6 +99,25 @@ const ChecklistResults: FC<ChecklistResultsProps> = ({
     window.print()
   }
 
+  // handle expanding all task details on print and closing tasks that weren't previously open after print
+  useEffect(() => {
+    window.matchMedia('print').addEventListener('change', (event) => {
+      if (event.matches) {
+        let details = [...document.querySelectorAll('details:not([open])')] as HTMLElement[]
+        for (let e of details) {
+          e.setAttribute('open', '')
+          e.dataset.wasclosed = ''
+        }
+      } else {
+        let details = [...document.querySelectorAll('details[data-wasclosed]')] as HTMLElement[]
+        for (let e of details) {
+          e.removeAttribute('open')
+          delete e.dataset.wasclosed
+        }
+      }
+    })
+  }, [])
+
```

Comment:
You forgot to `Clean up the event listener when the component unmounts`

```suggestion
  useEffect(() => {
    const mediaQuery = window.matchMedia('print')

    const handleChange = (event: MediaQueryListEvent) => {
      if (event.matches) {
        const detailsElements = [...document.querySelectorAll<HTMLDetailsElement>('details:not([open])')]
        detailsElements.forEach((el) => {
          el.open = true
          el.dataset.wasclosed = ''
        })
      } else {
        const detailsElements = [...document.querySelectorAll<HTMLDetailsElement>('details[data-wasclosed]')]
        detailsElements.forEach((el) => {
          el.open = false
          delete el.dataset.wasclosed
        })
      }
    }

    // Add the event listener
    mediaQuery.addEventListener('change', handleChange)

    // Clean up the event listener when the component unmounts
    return () => {
      mediaQuery.removeEventListener('change', handleChange)
    }
  }, [])
```